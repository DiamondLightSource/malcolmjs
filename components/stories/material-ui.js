import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import AppBar from 'material-ui/AppBar';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/RaisedButton';
import Toggle from 'material-ui/Toggle';
import { storiesOf, action } from '@kadira/storybook';

function handleEdgeDeleteButton(){

  console.log("material-ui button clicked");

}

storiesOf('Button', module)
    .add('material-ui flat button', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(lightBaseTheme)}>

            <FlatButton key="edgeDeleteButton" className="pull-right btn btn-default" label="edge Delete Button"
                          onClick={handleEdgeDeleteButton()}>Delete Edge</FlatButton>
        </MuiThemeProvider>
  ))
    .add('material-ui raised button', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(darkBaseTheme)}>
            <RaisedButton key="edgeDeleteButton" className="pull-right btn btn-default" label="edgeDeleteButton"
                          onClick={action("Raised button clicked")}>Delete Edge</RaisedButton>
        </MuiThemeProvider>
    ))
    .add('material-ui Toggle switch', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(darkBaseTheme)}>
            <Toggle key="edgeToggleSwitch" className="pull-right btn btn-default" label="Edge Toggle"
                          onToggle={action("Toggle switch")}/>
        </MuiThemeProvider>
    ));

