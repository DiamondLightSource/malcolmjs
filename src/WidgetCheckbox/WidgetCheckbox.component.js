import * as React from 'react';
import PropTypes from 'prop-types';
import Checkbox from 'material-ui/Checkbox';
import { withStyles } from 'material-ui/styles';
import { CircularProgress } from 'material-ui/Progress';
import { darken, lighten } from 'material-ui/styles/colorManipulator';

const styles = theme => ({
  div: {
    backgroundColor: theme.palette.background.paper,
  },
  checkBoxRoot: {
    color: lighten(theme.palette.primary.light, 0.1),
    '&$checkBoxChecked': {
      color: theme.palette.primary.light,
    },
  },
  checkBoxChecked: {},

  spinner: {
    size: 44,
    color: darken(theme.palette.primary.light, 0.25),
  },
});

const WidgetCheckbox = props => (
  <div className={props.classes.div}>
    {props.Pending ? (
      <CircularProgress size={44} className={props.classes.spinner} />
    ) : (
      <Checkbox
        checked={props.CheckState}
        onChange={props.checkEventHandler}
        classes={{
          root: props.classes.checkBoxRoot,
          checked: props.classes.checkBoxChecked,
        }}
        value={props.Label}
      />
    )}
  </div>
);

WidgetCheckbox.propTypes = {
  CheckState: PropTypes.bool.isRequired,
  checkEventHandler: PropTypes.func.isRequired,
  Pending: PropTypes.bool,
  Label: PropTypes.string.isRequired,
  classes: PropTypes.shape({
    div: PropTypes.string,
    checkBoxRoot: PropTypes.string,
    checkBoxChecked: PropTypes.string,
    spinner: PropTypes.string,
  }).isRequired,
};

WidgetCheckbox.defaultProps = {
  Pending: false,
};

export default withStyles(styles, { withTheme: true })(WidgetCheckbox);
