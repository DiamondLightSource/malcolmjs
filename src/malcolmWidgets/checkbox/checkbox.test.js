import React from 'react';
import { createShallow, createMount } from 'material-ui/test-utils';
import WidgetCheckbox from './checkbox.component';

describe('WidgetCheckbox', () => {
  let shallow;
  let mount;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('renders unchecked', () => {
    const wrapper = shallow(
      <WidgetCheckbox
        CheckState={false}
        checkEventHandler={() => {}}
        Label="TestBox"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls check', () => {
    const checkAction = jest.fn();
    const wrapper = mount(
      <WidgetCheckbox
        CheckState={false}
        checkEventHandler={() => {
          checkAction();
        }}
        Label="TestBox"
      />
    );

    wrapper.find('input').simulate('change');

    expect(checkAction.mock.calls.length).toEqual(1);
  });

  it('renders checked', () => {
    const wrapper = shallow(
      <WidgetCheckbox CheckState checkEventHandler={() => {}} Label="TestBox" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls uncheck', () => {
    const uncheckAction = jest.fn();
    const wrapper = mount(
      <WidgetCheckbox
        CheckState
        checkEventHandler={() => {
          uncheckAction();
        }}
        Label="TestBox"
      />
    );

    wrapper.find('input').simulate('change');

    expect(uncheckAction.mock.calls.length).toEqual(1);
  });
});
