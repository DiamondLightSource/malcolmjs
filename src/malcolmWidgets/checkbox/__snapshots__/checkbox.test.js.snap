// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetCheckbox renders checked 1`] = `
<WithStyles(Checkbox)
  checked={true}
  classes={
    Object {
      "checked": "WidgetCheckbox-checkBoxChecked-3",
      "root": "WidgetCheckbox-checkBoxRoot-2",
    }
  }
  disabled={false}
  indeterminate={false}
  onChange={[Function]}
  value="TestBox"
/>
`;

exports[`WidgetCheckbox renders unchecked 1`] = `
<WithStyles(Checkbox)
  checked={false}
  classes={
    Object {
      "checked": "WidgetCheckbox-checkBoxChecked-3",
      "root": "WidgetCheckbox-checkBoxRoot-2",
    }
  }
  disabled={false}
  indeterminate={false}
  onChange={[Function]}
  value="TestBox"
/>
`;
